solr/core/src/java/org/apache/solr/cloud/api/collections/SplitShardCmd.java

```
@@ -853,45 +857,61 @@ public static void checkDiskSpace(
-    SolrResponse rsp =
+        new ModifiableSolrParams().add("name", indexSizeMetric).add("name", freeDiskSpaceMetric);
+
+    var req =
         new GenericSolrRequest(
-                SolrRequest.METHOD.GET, "/admin/metrics", SolrRequest.SolrRequestType.ADMIN, params)
-            .process(cloudManager.getSolrClient());
+            SolrRequest.METHOD.GET, "/admin/metrics", SolrRequest.SolrRequestType.ADMIN, params);
+    req.setResponseParser(new InputStreamResponseParser("prometheus"));
+
+    SolrResponse resp = req.process(cloudManager.getSolrClient());
+
+    double[] sizes = new double[] {-1.0, -1.0}; // [indexSize, freeSize]
+    try (InputStream prometheusStream = (InputStream) resp.getResponse().get(STREAM_KEY);
+        var lines = NodeValueFetcher.Metrics.prometheusMetricStream(prometheusStream)) {
+
+      lines
+          .filter(line -> !line.isBlank() && !line.startsWith("#"))
+          .forEach(
+              line -> {
+                if (line.contains(indexSizeMetric) && line.contains(coreLabel)) {
+                  sizes[0] = NodeValueFetcher.Metrics.extractPrometheusValue(line);
+                } else if (line.contains(freeDiskSpaceMetric) && line.contains("usable_space")) {
+                  sizes[1] = NodeValueFetcher.Metrics.extractPrometheusValue(line);
+                }
+              });
+    }
+
+    double indexSize = sizes[0];
+    double freeSize = sizes[1];
 
-    Number size = (Number) rsp.getResponse()._get(List.of("metrics", indexSizeMetricName), null);
-    if (size == null) {
+    if (indexSize == -1.0) {
       log.warn("cannot verify information for parent shard leader");
       return;
     }
-    double indexSize = size.doubleValue();
 
-    Number freeSize =
-        (Number) rsp.getResponse()._get(List.of("metrics", freeDiskSpaceMetricName), null);
-    if (freeSize == null) {
+    if (freeSize == -1.0) {
       log.warn("missing node disk space information for parent shard leader");
       return;
     }
```

I'm thinking for here, this should throw instead of just return and continuing with the shard split cmd. Tests were passing because of this. Then we could at least know there was a regression. Was this just return for some specific purpose?

---

As a general rule, never log AND throw.  Pick one.  Here, you choose to throw, so ditch the log.  Logging is typically redundant with throwing, since throwing basically always gets logged as well.